### PR TITLE
Improve access logic and URL structure

### PR DESCRIPTION
**Pull request by odkhang**
This PR related to issue https://github.com/fossasia/eventyay-video/issues/240 Improve access logic and URL structure
It implement point 2 + 3 of the issue (point 1 is implemented by update nginx)

New button in event page of common, will redirect to video system with admin permission

image

## Summary by Sourcery

Enhance the event page by adding a button for video system access with admin permissions and improve the access logic by implementing a VideoAccessAuthenticator class to manage permissions and token generation.

New Features:
- Introduce a new button on the event page that redirects to the video system with admin permissions.

Enhancements:
- Improve the access logic by adding a VideoAccessAuthenticator class to handle video access permissions and token generation.

### CODE REVIEW

File: src/pretix/eventyay_common/tasks.py [hunk 1]
```diff
@@ -10,100 +10,153 @@
...
+    @self: task instance
+    @param is_video_creation: allow user to add video system
+    @param data: event's data
+    """
+    event_slug = data.get("id")
+    title = data.get("title")
+    event_timezone = data.get("timezone")
+    locale = data.get("locale")
+    token = data.get("token")
+    has_permission = data.get("has_permission")
```
Comment (hongquan) [inline src/pretix/eventyay_common/tasks.py]: Please make `token`, `has_permission` explicit parameters of `create_world` function.

File: src/pretix/eventyay_common/tasks.py [hunk 2]
```diff
@@ -10,100 +10,153 @@
...
+    # Check if user choose add video option and has permission to create video system ('can_create_events' permission)
+    if is_video_creation and has_permission:
+        try:
+            requests.post(
+                "{}/api/v1/create-world/".format(settings.VIDEO_SERVER_HOSTNAME),
+                json=payload,
+                headers=headers,
+            )
+        except requests.exceptions.ConnectionError as e:
+            logger.error(f"Connection error: {e}")
```
Comment (hongquan) [inline src/pretix/eventyay_common/tasks.py]: Wrong use of `logger`. This has been remined many times.

File: src/pretix/eventyay_common/utils.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+logger = logging.getLogger(__name__)
+
+
+def generate_token(request):
+    """
+    Generate token for video system
+    @param request: user request
+    @return: jwt
+    """
+    uid_token = encode_email(request.user.email)
```
Comment (hongquan) [inline src/pretix/eventyay_common/utils.py]: Why `email` need to be encoded?

**Review by hongquan — CHANGES_REQUESTED** [review]

File: src/pretix/eventyay_common/utils.py (hunk 3 above)
Comment (odkhang) [reply src/pretix/eventyay_common/utils.py]: I would like to create a unique identification for user authentication in the video system.

**Review by hongquan — APPROVED** [review]

**Pull request merged**